libs/core/src/components/pds-button/pds-button.tsx

```diff
@@ -133,6 +136,26 @@ export class PdsButton {
     return classNames.join(' ');
   }
 
+  private renderStartContent() {
+    if (this.icon && this.variant !== 'disclosure') {
+      return (
+        <pds-icon class={this.loading ? 'pds-button__icon--hidden' : ''} name={this.icon} part="icon" aria-hidden="true"></pds-icon>
+      );
+    } else {
+      return <slot name="start" />;
+    }
```

Comment:
I understand it may be a breaking change, but if we are moving to slots, we should deprecate the icon prop. Maybe we start by removing it from the docs (OR update any examples with icons to use the slot), and then once we've updated KP usage, we could do away with the prop from the code.

Just a thought, but in my opinion, we should only have one or the other to reduce confusion. Definitely check with the others.

Reply:
Yep, deprecation will come in a future update. I've updated the description to reflect that and also updated the docs examples to use slots.